From akpm@osdl.org Wed Jan 18 23:55:40 2006
Message-Id: <200601190755.k0J7tLRO027507@shell0.pdx.osdl.net>
From: Andrew Morton <akpm@osdl.org>
Subject: USB: optimise devio.c usbdev_read fix
To: greg@kroah.com
Cc: akpm@osdl.org, gregkh@suse.de, oliver@neukum.org, zaitcev@redhat.com
From: akpm@osdl.org
Date: Wed, 18 Jan 2006 23:55:07 -0800


From: Andrew Morton <akpm@osdl.org>

drivers/usb/core/devio.c: In function `usbdev_read':
drivers/usb/core/devio.c:140: error: invalid type argument of `->'
drivers/usb/core/devio.c:141: error: invalid type argument of `->'
drivers/usb/core/devio.c:142: error: invalid type argument of `->'
drivers/usb/core/devio.c:143: error: invalid type argument of `->'

Cc: Oliver Neukum <oliver@neukum.org>
Cc: Pete Zaitcev <zaitcev@redhat.com>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/usb/core/devio.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

--- gregkh-2.6.orig/drivers/usb/core/devio.c
+++ gregkh-2.6/drivers/usb/core/devio.c
@@ -137,10 +137,10 @@ static ssize_t usbdev_read(struct file *
 		struct usb_device_descriptor temp_desc ; /* 18 bytes - fits on the stack */
 
 		memcpy(&temp_desc, &dev->descriptor, sizeof(dev->descriptor));
-		le16_to_cpus(&temp_desc->bcdUSB);
-		le16_to_cpus(&temp_desc->idVendor);
-		le16_to_cpus(&temp_desc->idProduct);
-		le16_to_cpus(&temp_desc->bcdDevice);
+		le16_to_cpus(&temp_desc.bcdUSB);
+		le16_to_cpus(&temp_desc.idVendor);
+		le16_to_cpus(&temp_desc.idProduct);
+		le16_to_cpus(&temp_desc.bcdDevice);
 
 		len = sizeof(struct usb_device_descriptor) - pos;
 		if (len > nbytes)
